tests/test_filtsmooth/test_statespace/test_discrete_transition.py

```diff
@@ -234,6 +240,83 @@ def test_all_backward_rv_same_with_cache(
         np.testing.assert_allclose(out_joseph.mean, out_sqrt.mean)
         np.testing.assert_allclose(out_joseph.cov, out_sqrt.cov)
 
+    def test_all_backward_realization_same_no_cache(
+        self, some_normal_rv1, some_normal_rv2, diffusion
+    ):
+        """Assert all implementations give the same output -- no gain or forwarded RV
+        passed."""
+
+        out_classic, _ = self.transition._backward_rv_classic(
+            pnrv.Constant(some_normal_rv1.mean),
```

Comment:
these tests would have failed with the old implementation (I checked).

Reply:
Very nice! Great that you put in the effort to add such tests in there :)